index.js

```diff
@@ -458,10 +458,13 @@ function normalizeArguments(url, opts) {
 		opts
 	);
 
+	const headers = lowercaseKeys(opts.headers);
+	Object.keys(headers).forEach(key => (headers[key] === null || headers[key] === undefined) && delete headers[key]);
```

Comment:
There's a lot going on in this line and it's inconsistent with the rest of the codebase.

e.g `forEach` instead of `for...of`, `&&` instead of `if`.

I would suggest the following for better readability and consistency.

```js
for (const key of Object.keys(headers)) {
  if (headers[key] === null || headers[key] === undefined) {
    delete headers[key];
  }
}
```

Functionally all looks good though 👍